lib/readline.js

```diff
@@ -965,91 +975,124 @@ const escapeCodeReAnywhere = new RegExp([
   functionKeyCodeReAnywhere.source, metaKeyCodeReAnywhere.source, /\x1b./.source
 ].join('|'));
 
-function emitKeys(stream, s) {
-  if (s instanceof Buffer) {
-    if (s[0] > 127 && s[1] === undefined) {
-      s[0] -= 128;
-      s = '\x1b' + s.toString(stream.encoding || 'utf-8');
-    } else {
-      s = s.toString(stream.encoding || 'utf-8');
-    }
-  }
```

Comment:
No.

I'd appreciate another look at this, but I think this `Buffer` handling was a dead code since this commit: https://github.com/iojs/io.js/commit/3c91a7ae10f0ccabe4550c77189813f8d95785b0.

This function never receives any buffers in the first place, because [string_decoder.write()](https://github.com/rlidwka/io.js/blob/0450ce7db22ab4b6f9b2119197389ed7d3eac8c3/lib/string_decoder.js#L62) returns strings only.